### PR TITLE
Fix open issues on Patterns article

### LINKED ISSUE #26855: Add example/documentation for 'is not' with non-nullable declaration pattern
Please add an example/section in the '[Declaration and type patterns](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#declaration-and-type-patterns)' section to showcase the use of a non-nullable declaration pattern with 'is not' use case, which adds the equivalent functionality of Swift's `guard let`, [discussed in feature request here](https://github.com/dotnet/csharplang/discussions/138#discussioncomment-1601167).

```
if (someObject.SomeProperty is not string someProperty) {
    return; // or 'continue'
}
```

While the positive case is covered with this example: `if (greeting is string message) { .. }`, the inverse is non-obvious and opens up very useful null-safe short-circuiting control flows that are not stated anywhere in the documentation. The only current example is `if (input is not null) { .. }` in the '[Constant pattern](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#constant-pattern)' and '[Logical patterns](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#logical-patterns)' sections, but that does not combine a [Declaration pattern](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#declaration-and-type-patterns) within it – which is what brings all the extra usefulness.

The existence of this feature is a major component of writing modern, null-safe, idiomatic C#, supported since C# 9, let's get the word out about it!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9112c305-f1cd-1aa5-4557-2c7d757e6ca0
* Version Independent ID: 312e9c81-ee2a-1d84-64d7-bcaa71d7eab8
* Content: [Patterns - C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#declaration-and-type-patterns)
* Content Source: [docs/csharp/language-reference/operators/patterns.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/operators/patterns.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #29242: Patterns (C# Reference): List patterns broken sentence
The section on *list patterns* ends with a weird sentence:

> You any other pattern to match characteristics about individual elements.

I guess the intended sentence would be

> You *can use* any other pattern to match characteristics about individual elements.

It's probably also a good idea to show what this looks like, such as

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9112c305-f1cd-1aa5-4557-2c7d757e6ca0
* Version Independent ID: 312e9c81-ee2a-1d84-64d7-bcaa71d7eab8
* Content: [Patterns - C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns)
* Content Source: [docs/csharp/language-reference/operators/patterns.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/operators/patterns.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #29242 (type)
Fixes #26855 (Add not null pattern)

### CODE REVIEW

File: docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePatterns.cs [hunk 1]
```diff
@@ -7,8 +7,22 @@ public static void Examples()
             BasicExample();
             ReferenceConversion();
             NullableAndUnboxing();
+            NotNullPattern();
         }
 
+        private static void NotNullPattern()
+        {
+            object input;
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePatterns.cs:15]: build issue:

```suggestion
            object input = default;
```

File: docs/csharp/language-reference/operators/patterns.md [hunk 2]
```diff
@@ -70,6 +70,10 @@ Beginning with C# 9.0, for that purpose you can use a *type pattern*, as the fol
 
 Like a declaration pattern, a type pattern matches an expression when an expression result is non-null and its run-time type satisfies any of the conditions listed above.
 
+You can also this pattern for a clear, concise `null` check:
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/patterns.md:73]:
```suggestion
You can also use this pattern for a clear, concise `null` check:
```

File: docs/csharp/language-reference/operators/patterns.md [hunk 3]
```diff
@@ -294,7 +298,7 @@ It's a CSV format, but some of the rows have more columns than others. Even wors
 
 The preceding example takes a string array, where each element is one field in the row. The switch expression keys on the second field, which determines the kind of transaction, and the number of remaining columns. Each row ensures the data is in the correct format. The discard pattern (`_`) skips the first field, with the date of the transaction. The second field matches the type of transaction. Remaining element matches skip to the field with the amount. The final match uses the *var* pattern to capture the string representation of the amount. The expression calculates the amount to add or subtract from the balance.
 
-*List patterns* enable you to match on the shape of a sequence of data elements. You use the *discard* and *range* patterns to match the location of elements. You any other pattern to match characteristics about individual elements.
+*List patterns* enable you to match on the shape of a sequence of data elements. You use the *discard* and *range* patterns to match the location of elements. You use other pattern to match characteristics about individual elements.
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/patterns.md:301]:
```suggestion
*List patterns* enable you to match on the shape of a sequence of data elements. You use the *discard* and *range* patterns to match the location of elements. You use other patterns to match characteristics about individual elements.
```

**Review by gewarren — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @gewarren 

Interesting that Acrolinx didn't flag either of the two issues you noted.

**Pull request merged**